### DISCUSSION TITLE
How to install peacock

### DISCUSSION THREAD

**Discussion by yuegrit** [idaholab/moose#28074] [Q&A Getting Started]
### Check these boxes if you have followed the posting rules.

- [X] I have consulted the [Posting Guidelines](https://github.com/idaholab/moose/discussions/18270).
- [X] I have searched the [Discussions Forum](https://github.com/idaholab/moose/discussions) and [MOOSE Framework Troubleshooting](https://mooseframework.inl.gov/help/troubleshooting.html) and have not found what I was looking for
- [X] Q&A Getting Started is the most appropriate category for my question (trouble installing, beginner user, ...)

### Issue or question about MOOSE

Hello, everyone. I was confronted with a problem when I installed Peacock into Ubantu according to this instruction https://mooseframework.inl.gov/python/peacock.html. After I inputted command "conda create -n peacock moose-peacock", a error happened like this:
(base) yue@yue-laptop:~$ conda create -n peacock moose-peacock
Collecting package metadata (current_repodata.json): done
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: \ Killed

Is there anyone that could help me with this error ? Thank you very much.

### Encountering Errors? Please include diagnostic output

```pre
(base) yue@yue-laptop:~$ conda create -n peacock moose-peacock
Collecting package metadata (current_repodata.json): done
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: \ Killed
```

**Comment by GiudGiud** [comment]
Hello

Killed likely means you ran out of memory, unless you killed the process yourself with ctrl C

Guillaume

**Reply by yuegrit** [reply]
Thank you very much. This time, I used a linux-based computer with a memory of 32 Gb according to your suggestion. However, a new problem appear like this:
[cloudam@master ~]$ conda create -n peacock moose-peacock
Channels:
 - https://conda.software.inl.gov/public
 - anaconda
 - intel
 - defaults
Platform: linux-64
Collecting package metadata (repodata.json): done
Solving environment: failed

LibMambaUnsatisfiableError: Encountered problems while solving:
  - nothing provides _sysroot_linux-64_curr_repodata_hack 3 h69a702a_13 needed by moose-mpich-4.0.2-build_15

Could not solve for environment specs
The following packages are incompatible
└─ moose-peacock is not installable because there are no viable options
   ├─ moose-peacock 2022.10.04 would require
   │  └─ ca-certificates 2022.9.24 ha878542_0, which does not exist (perhaps a missing channel);
   ├─ moose-peacock 2022.12.05 would require
   │  └─ ca-certificates 2022.12.7 ha878542_0, which does not exist (perhaps a missing channel);
   ├─ moose-peacock 2023.04.11 would require
   │  └─ moose-mpich 4.0.2 build_16, which requires
   │     └─ _sysroot_linux-64_curr_repodata_hack 3 h69a702a_14, which does not exist (perhaps a missing channel);
   ├─ moose-peacock [2023.08.17|2023.08.31|...|2023.11.10] would require
   │  └─ paraview, which does not exist (perhaps a missing channel);
   └─ moose-peacock 2023.11.29 would require
      └─ moose-mpich 4.0.2 build_15, which requires
         └─ _sysroot_linux-64_curr_repodata_hack 3 h69a702a_13, which does not exist (perhaps a missing channel).

Could you please help with this problem?

**Reply by GiudGiud** [reply]
@milljm may know what problem we are having with package dependencies here

**Reply by milljm** [reply]
Our packages require the [conda-forge](https://conda-forge.org/) channel to be made available. And to be strictly used with other packages coming from conda-forge (no mix-and-matching).

This installation looks to be of a `miniconda` variant (denoting the Anaconda channel there). You can try the following, but it might be necessary to install Miniforge which makes direct use of the 'conda-forge' channel.


```bash
conda config --add channels conda-forge
conda config --set channel_priority strict
conda create -n peacock moose-peacock
```

**Reply by milljm** [reply]
Also, `miniconda` is going to be nefariously slow at solving dependencies. This right here:

```pre
Solving environment: 
```
Prepare for this to take some times 40 minutes. It is not broken!

This is why we primarily moved to [Miniforge](https://conda-forge.org/download/) instead (their solver is parallel/threaded, solving dependency chains in a fraction of the time).
